native/spark-expr/src/conversion_funcs/cast.rs

```diff
@@ -1528,6 +1529,29 @@ where
     Ok(Arc::new(output_array))
 }
 
+fn spark_cast_decimal_to_boolean(
+    array: &dyn Array,
+    from_type: &DataType,
+    to_type: &DataType,
```

Comment:
This function is specifically for casting decimal to boolean, so I don't think we need to pass in the types since they were checked before calling this function.